tensorflow_datasets/scripts/replace_fake_images.py

```diff
@@ -104,20 +101,24 @@ def rewrite_tar(root_dir, tar_filepath):
     # Checking the extension of file to be extract
     if tar_filepath.lower().endswith('gz'):
       extension = ':gz'
+      shutil_extension = 'gztar'
+      extension_removal_index = 7 # Hardcode number of indices to be skipped from the end to obtain filename. This is hard-coded as filenames containing "." might exist and thus filepath.split(".") would give an incorrect result.
```

Comment:
Rather than hardcoding a number, I think it would be more explicit to use `len('.tar.gz')`, `len('.zip')`,...